### PR TITLE
feat: support sanitized kernel args

### PR DESCRIPTION
**Pull request by frezbo**
Support dropping kernel args that start with `-`.

Fixes: #7613

### CODE REVIEW

File: cmd/installer/pkg/install/errata.go [hunk 1]
```diff
@@ -101,3 +103,28 @@ func upgradeFromPreIfnamesTalos() (bool, error) {
...
+
+// SanitizeKernelArgs removes kernel arguments that are not needed for the installer.
+func SanitizeKernelArgs(cmdline []string) []string {
+	kernelArgsToDrop := xslices.Filter(cmdline, func(s string) bool {
+		// it's valid to have a kernel arg just as `-`.
+		// Ref: https://www.kernel.org/doc/html/v4.14/admin-guide/kernel-parameters.html
+		return len(s) > 1 && s[0] == '-'
+	})
+
+	return xslices.Filter(cmdline, func(s string) bool {
```
Comment (smira) [inline cmd/installer/pkg/install/errata.go:115]: would it make more sense to put this into the `cmdline.AppendAll` code?

Comment (frezbo) [reply cmd/installer/pkg/install/errata.go:115]: yeh, that's fine, I was not sure if we should do that there

**Review by smira — APPROVED** [review]
Can you please add an example to the Boot Assets documentation?

And probably machine config extra kernel args?

**Pull request merged**